### PR TITLE
lib: modem_key_mgmt: use new AT APIs

### PR DESCRIPTION
**Pull request by lemrey**
- update `modem_key_mgmt` to use new AT APIs
- remove dependency on AT parser, lower heap requirements
- remove unsupported function (`_permission_set()`) and parameter (`perm_flags` in `_exists()`)
- bump some projects main stack sizes due to increased usage from `scanf` and other C stdlib functions used in the new AT APIs

### CODE REVIEW

**Review by rlubos — APPROVED** [review]
LGTM. Do we need changelog entry for that?

File: lib/modem_key_mgmt/modem_key_mgmt.c [hunk 1]
```diff
@@ -7,96 +7,77 @@
...
 {
 	int err;
-	char response[sizeof("+CMEE: X\r\n")];
+	int active;
 
-	err = at_cmd_write("AT+CMEE?", response, sizeof(response), NULL);
-	if (err) {
+	err = nrf_modem_at_scanf("AT+CMEE?", "+CMEE: %d", &active);
+	if (err < 0) {
 		return err;
```
Comment (tokangas) [inline lib/modem_key_mgmt/modem_key_mgmt.c:30]: The function was changed to return a boolean, but tries to return an error here.

**Review by tokangas — COMMENTED** [review]
Looks good to me, only one finding.

**Comment by ChristopherBittner** [comment]
Failing CI for this PR was rerun on a special branch as it would be failing with master of tests.
Therefor red status can be ignored for now.

File: lib/modem_key_mgmt/modem_key_mgmt.c (hunk 1 above)
Comment (lemrey) [reply lib/modem_key_mgmt/modem_key_mgmt.c]: Fixed

**Comment by lemrey** [comment]
Addressed @tokangas comment, please have another look :)
@jtguggedal wanna have a look too?

**Review by simensrostad — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]
Looks good to me, thanks!

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Why is the manifest being updated in the same commit as `lib: modem_key_mgmt: remove unused parameter in _exists()`?

This should be its own commit, or the commit log should be updated to explain why the change is being done, not just explain the change itself.

**Comment by mbolivar-nordic** [comment]
@lemrey If you push a zero-diff change fixing the log or splitting up the commits, I will merge it without waiting for CI again since it's green.

**Comment by lemrey** [comment]
@mbolivar-nordic I have reworded that commit message.

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**